### PR TITLE
rework LAN display and service launchability

### PR DESCRIPTION
**Pull request by MattDHill**

### CODE REVIEW

File: ui/src/app/models/app-types.ts [hunk 1]
```diff
@@ -37,18 +37,23 @@ export interface AppInstalledPreview extends BaseApp {
   lanAddress?: string
   torAddress: string
   versionInstalled: string
-  ui: boolean
+  lanUi: boolean
+  torUi: boolean
```
Comment (ProofOfKeags) [inline ui/src/app/models/app-types.ts:41]: I've updated the agent to reflect this.

File: ui/src/app/pages/apps-routes/app-installed-list/app-installed-list.page.ts [hunk 2]
```diff
@@ -35,19 +35,15 @@ export class AppInstalledListPage extends Cleanup {
   segmentValue: 'services' | 'embassy' = 'services'
 
   showCertDownload : boolean
-  isConsulate: boolean
```
Comment (ProofOfKeags) [inline ui/src/app/pages/apps-routes/app-installed-list/app-installed-list.page.ts:38]: is this just cleanup or is it a direct result of this change?

File: ui/src/app/pages/apps-routes/app-installed-show/app-installed-show.page.ts [hunk 3]
```diff
@@ -64,8 +62,12 @@ export class AppInstalledShowPage extends Cleanup {
     this.cleanup(
       markAsLoadingDuring$(this.$loading$, this.preload.appFull(this.appId))
         .pipe(
-          tap(app => this.app = app),
-          concatMap(() => this.syncWhenDependencyInstalls()), //must be final in stack
+          tap(app => {
+            this.app = app
+            const appP = peekProperties(this.app)
+            this.hideLAN = !appP.lanAddress || (appP.id === 'mastodon' && appP.versionInstalled === '3.3.0') // @TODO delete this hack in 0.3.0
```
Comment (ProofOfKeags) [inline ui/src/app/pages/apps-routes/app-installed-show/app-installed-show.page.ts:68]: is this the only version, or are subsequent emver revisions also affected?

File: ui/src/app/services/api/mock-app-fixures.ts [hunk 4]
```diff
@@ -57,23 +61,23 @@ export const bitcoinI: AppInstalledFull = {
   instructions: 'some instructions',
   lastBackup: new Date().toISOString(),
   configuredRequirements: [],
-  hasFetchedFull: true,
-  ui: false,
+  lanUi: false,
+  torUi: false,
```
Comment (ProofOfKeags) [inline ui/src/app/services/api/mock-app-fixures.ts:65]: as far as I can tell the API has never returned ui available on AppInstalledFull. We can amend it to but that isn't how it works right now.

File: ui/src/app/services/api/mock-app-fixures.ts [hunk 5]
```diff
@@ -57,23 +61,23 @@ export const bitcoinI: AppInstalledFull = {
...
   actions: [
     { id: 'sync-chain', name: 'Sync Chain', description: 'this will sync with the chain like from Avatar', allowedStatuses: [ AppStatus.RUNNING, AppStatus.RUNNING, AppStatus.RUNNING, AppStatus.RUNNING ]},
     { id: 'off-sync-chain', name: 'Off Sync Chain', description: 'this will off sync with the chain like from Avatar', allowedStatuses: [ AppStatus.STOPPED ]}
   ],
 }
 
-export const lightningI: AppInstalledFull = {
-  id: 'c-lightning',
+export const lightningI: ApiAppInstalledFull = {
+  id: 'lightning',
```
Comment (ProofOfKeags) [inline ui/src/app/services/api/mock-app-fixures.ts:74]:
```suggestion
  id: 'c-lightning',
```

File: ui/src/app/pages/apps-routes/app-installed-list/app-installed-list.page.ts (hunk 2 above)
Comment (MattDHill) [reply ui/src/app/pages/apps-routes/app-installed-list/app-installed-list.page.ts:38]: clenaup

File: ui/src/app/pages/apps-routes/app-installed-show/app-installed-show.page.ts (hunk 3 above)
Comment (MattDHill) [reply ui/src/app/pages/apps-routes/app-installed-show/app-installed-show.page.ts:68]: Subsequent versions should have the proper `lanUi` boolean, so only needed for 0.3.0.

**Review by ProofOfKeags — APPROVED** [review]

**Pull request merged**